infra/style/format.sh

```diff
@@ -6,13 +6,42 @@ set -e
 APPLY_PATCH_OPTION="-a"
 while [[ "$#" -gt 0 ]]; do
   case $1 in
-    --no-apply-patches) APPLY_PATCH_OPTION="";;
+    --no-apply-patches)
+      APPLY_PATCH_OPTION=""
+      shift
+      ;;
+    --diff-only)
+      CHECK_DIFF_ONLY="1"
+      shift
+      ;;
     *) echo "[ERROR] Unknown parameter passed: $1"; exit 255 ;;
   esac
-  shift
 done
 
 GIT_ROOT="$(git rev-parse --show-toplevel)"
 STYLE_DIR="${GIT_ROOT}/infra/style"
 
-lintrunner --force-color --all-files $APPLY_PATCH_OPTION --config "${GIT_ROOT}/.lintrunner.toml"
+if [[ "${CHECK_DIFF_ONLY}" = "1" ]]; then
+  MAIN_EXIST=$(git rev-parse --verify main)
+  CURRENT_BRANCH=$(git branch | grep \* | cut -d ' ' -f2-)
+  DIFF_COMMITS=`git log --graph --oneline main..HEAD | wc -l`
+  if [[ -z "${MAIN_EXIST}" ]]; then
+    echo "Cannot find main branch"
```

Comment:
Question, 
is it okay to not exit when the `main` branch does not exist?

```suggestion
    echo "Cannot find main branch"
    exit 1
```

Reply:
I've updated the code. Thanks!